datumaro/plugins/synthetic_images_plugin/image_generator.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from multiprocessing import Pool
+from shutil import rmtree
+from typing import List, Optional
+import os
+
+import cv2 as cv
+import numpy as np
+
+from datumaro.components.cli_plugin import CliPlugin
+from datumaro.components.errors import SourceExistsError
+
+from .utils import IFSFunction, augment, colorize, download_colorization_model
+
+
+class ImageGenerator(CliPlugin):
+    """
+    ImageGenerator generates a 3-channel synthetic dataset with provided shapes.
+    """
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('-k', '--count', type=int, required=True,
+            help="Number of images to generate")
+        parser.add_argument('-o', '--output-dir', type=os.path.abspath, required=True,
+            help="Path to the directory where dataset are saved")
+        parser.add_argument('--shape', nargs='+', type=int,
+            help="Image shape: height and width separated by a space, "
+                 "for example --shape 256 224")
+        parser.add_argument('--overwrite', action='store_true',
+            help="Overwrite existing files in the save directory")
+
+        return parser
+
+    def __init__(self, count: int, output_dir: str,
+                 shape: List[int], overwrite: bool = False):
+        self._count = count
+        self._output_dir = output_dir
+        assert len(shape) == 2
+        self._height, self._width = shape
+
+        if not os.path.exists(output_dir):
+            os.mkdir(output_dir)
+        elif os.path.isdir(output_dir) and os.listdir(output_dir):
+            if overwrite:
+                rmtree(output_dir)
+                os.mkdir(output_dir)
+            else:
+                raise SourceExistsError(output_dir)
```

Comment:
This exception type is unrelated to the condition you're reporting here. Please check commands that support the `--overwrite` option to see how this should be handled.

Reply:
Changed to `CliException`